xcm/src/v0/mod.rs

```diff
@@ -0,0 +1,176 @@
+	/// - `effects`: The order(s) to execute on the holding account.
+	///
+	/// Safety: `origin` must be trusted to have received and be storing `assets` such that they may later be
+	/// withdrawn should this system send a corresponding message.
+	///
+	/// Kind: *Trusted Indication*.
+	///
+	/// Errors:
+	ReserveAssetDeposit { assets: Vec<MultiAsset>, effects: Vec<Order> },
+
+	/// Asset(s) (`assets`) have been destroyed on the `origin` system and equivalent assets should be
+	/// created on this system.
+	///
+	/// Some orders are given (`effects`) which should be executed once the corresponding derivative assets have
+	/// been placed into `holding`.
+	///
+	/// - `assets`: The asset(s) that are minted into holding.
+	/// - `effects`: The order(s) to execute on the holding account.
+	///
+	/// Safety: `origin` must be trusted to have irrevocably destroyed the `assets` prior as a consequence of
+	/// sending this message.
+	///
+	/// Kind: *Trusted Indication*.
+	///
+	/// Errors:
+	TeleportAsset { assets: Vec<MultiAsset>, effects: Vec<Order> },
+
+	/// Indication of the contents of the holding account corresponding to the `QueryHolding` order of `query_id`.
+	///
+	/// - `query_id`: The identifier of the query that resulted in this message being sent.
+	/// - `assets`: The message content.
+	///
+	/// Safety: No concerns.
+	///
+	/// Kind: *Information*.
+	///
+	/// Errors:
+	Balances { #[codec(compact)] query_id: u64, assets: Vec<MultiAsset> },
+
+	/// Apply the encoded transaction `call`, whose dispatch-origin should be `origin` as expressed by the kind
+	/// of origin `origin_type`.
+	///
+	/// - `origin_type`: The means of expressing the message origin as a dispatch origin.
+	/// - `call`: The encoded transaction to be applied.
+	///
+	/// Safety: No concerns.
+	///
+	/// Kind: *Instruction*.
+	///
+	/// Errors:
+	Transact { origin_type: OriginKind, call: Vec<u8> },
+
+	/// Relay an inner message (`inner`) to a locally reachable destination ID `dest`.
+	///
+	/// The message sent to the destination will be wrapped into a `RelayedFrom` message, with the
+	/// `superorigin` being this location.
+	///
+	/// - `dest: MultiLocation`: The location of the to be relayed into. This may never contain `Parent`, and
+	///   it must be immediately reachable from the interpreting context.
+	/// - `inner: VersionedXcm`: The message to be wrapped and relayed.
+	///
+	/// Safety: No concerns.
+	///
+	/// Kind: *Instruction*.
+	///
+	/// Errors:
+	RelayTo { dest: MultiLocation, inner: Box<VersionedXcm> },
+
+	/// A message (`inner`) was sent to `origin` from `superorigin` with the intention of being relayed.
+	///
+	/// - `superorigin`: The location of the `inner` message origin, **relative to `origin`**.
+	/// - `inner`: The message sent by the super origin.
+	///
+	/// Safety: `superorigin` must express a sub-consensus only; it may *NEVER* contain a `Parent` junction.
+	///
+	/// Kind: *Trusted Indication*.
+	///
+	/// Errors:
```

Comment:
Maybe remove for now?

Reply:
I'd prefer to keep them if you don't mind, to save us work updating this now and later during rebasing while this doesn't seem to be actively harmful